src/pymor/analyticalproblems/functions.py

```diff
@@ -401,43 +402,97 @@ def evaluate(self, x, mu=None):
+                from PIL import Image
+            except ImportError as e:
+                raise ImportError("PIL is needed for loading images. Try 'pip install pillow'") from e
+            img = Image.open(filename)
+            if not img.mode == 'L':
+                self.logger.warning('Image ' + filename + ' not in grayscale mode. Converting to grayscale.')
+                img = img.convert('L')
+            bitmap = np.array(img).T[:, ::-1]
+            bitmap = bitmap * ((range[1] - range[0]) / 255.)  # copy to change dtype
+            bitmap += range[0]
+            filename = None
+            range = None
+        assert isinstance(bitmap, np.ndarray)
+        assert range is None, 'only for deprecated interface'
+        assert filename is None, 'only for deprecated interface'
         bounding_box = bounding_box or [[0., 0.], [1., 1.]]
+        self.__auto_init(locals())
+        self.lower_left = np.array(bounding_box[0])
+        self.size = np.array(bounding_box[1] - self.lower_left)
+
+    @classmethod
+    def from_file(cls, filename, bounding_box=None, range=None):
+        """Create a |BitmapFunction| from a grayscale image file.
+
+        Parameters
+        ----------
+        filename
+            Path of the image representing the function.
+        bounding_box
+            Lower left and upper right coordinates of the domain of the function.
+        range
+            A pixel of value p is mapped to `(p / 255.) * range[1] + range[0]`.
+        """
         range = range or [0., 1.]
         try:
             from PIL import Image
         except ImportError as e:
             raise ImportError("PIL is needed for loading images. Try 'pip install pillow'") from e
         img = Image.open(filename)
         if not img.mode == 'L':
-            self.logger.warning('Image ' + filename + ' not in grayscale mode. Converting to grayscale.')
+            cls._logger.warning('Image ' + filename + ' not in grayscale mode. Converting to grayscale.')
             img = img.convert('L')
-        self.__auto_init(locals())
-        self.bitmap = np.array(img).T[:, ::-1]
-        self.lower_left = np.array(bounding_box[0])
-        self.size = np.array(bounding_box[1] - self.lower_left)
+        bitmap = np.array(img).T[:, ::-1]
+        bitmap = bitmap * ((range[1] - range[0]) / 255.)  # copy to change dtype
+        bitmap += range[0]
+        return cls(bitmap, bounding_box=bounding_box)
+
+    @classmethod
+    def random(cls, shape=(1, 1), bounding_box=((0., 0.), (1., 1.)), range=(0., 1.)):
+        """Create a random |BitmapFunction| with uniform distribution of values.
+
+        Parameters
+        ----------
+        bounding_box
+            Lower left and upper right coordinates of the domain of the function.
+        range
+            Tuple of minimum and maximum functuion values.
+        """
+        a, b = range[0], range[1]
+        random_field = get_rng().uniform(a, b, np.prod(shape)).reshape(shape)
```

Comment:
Do we want to allow for different distributions?

Reply:
In principle yes, but maybe in a future PR ..